src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1284,7 +1423,65 @@ static void run_server(Preforker& forker, NBDServer *server, bool netlink_used)
   shutdown_async_signal_handler();
 }
 
-static int do_map(int argc, const char *argv[], Config *cfg)
+#ifndef __NR_pidfd_open
+#define __NR_pidfd_open 434   /* System call # on most architectures */
```

Comment:
Nit: if there is no glibc wrapper for it and it's not defined, I think you should just fallback to the `/proc/` approach ... or just always open the `/proc/<pid>` path to avoid the fallback path.